core/shared/src/main/scala/zio/Chunk.scala

```diff
@@ -498,6 +498,26 @@ sealed abstract class Chunk[+A] extends ChunkLike[A] { self =>
     exists
   }
 
+  /**
+   * Applies the side-effecting function `f` to the elements of this chunk for as long
+   * as it returns `true`.
+   */
+  def foreachWhile(f: A => Boolean): Unit = {
```

Comment:
By convention we should call this `unsafeForeachWhile` or make it package private to `zio`.